benchmarks/profilers/benchmark_batch.py

```diff
@@ -0,0 +1,284 @@
+    return ExecuteModelRequest(seq_group_metadata_list=seq_group_metadata_list,
+                               # All the rest stay as default
+                               )
+
+
+@lru_cache(maxsize=16384)
+def build_dummy_seq_group_metadata(
+    llm: AsyncLLMEngine,
+    tokenizer: PreTrainedTokenizerBase,
+    input_len: int,
+    is_prompt: bool,
+) -> SequenceGroupMetadata:
+    """
+    Send requests as new every time (no `SequenceGroupMetadataDelta`).
+    """
+    seq = SequenceData.from_seqs([
+        random.randint(0, tokenizer.vocab_size - 1) for _ in range(input_len)
+    ])
+    if not is_prompt:
+        seq.update_num_computed_tokens(input_len - 1)
+
+    seq_data: Dict[int, SequenceData] = {0: seq}
+
+    # Same as in `benchmark_throughput.py`
+    sampling_params = SamplingParams(
+        n=1,
+        temperature=1.0,
+        top_p=1.0,
+        ignore_eos=True,
+        max_tokens=2048,  # TODO: remove this hardcoded value
+    )
+
+    # Build a random block mapping
+    # TODO: try sequential block tables
+    block_manager = llm.engine.scheduler[0].block_manager
+    assert isinstance(block_manager, SelfAttnBlockSpaceManager)
+    block_size = block_manager.block_size
+    num_required_blocks = cdiv(input_len, block_size)
+    block_tables: Dict[int, List[int]] = {
+        0: [
+            random.randint(0, block_manager.num_total_gpu_blocks)
+            for _ in range(num_required_blocks)
+        ]
+    }
+
+    # For simplicity, assume all prefill and decode requires sampling. In
+    # practice, if prefill is chunked, only the last chunk requires sampling
+    do_sample = True
+
+    ret = SequenceGroupMetadata(
+        request_id=random_uuid(),
+        is_prompt=is_prompt,
+        seq_data=seq_data,
+        sampling_params=sampling_params,
+        block_tables=block_tables,
+        do_sample=do_sample,
+        # Assume the rest doesn't matter and uses defaults
+    )
+    return ret
+
+
+def get_stats(llm: AsyncLLMEngine, model_output: List[SamplerOutput]) -> Stats:
+    return llm.engine._get_stats(
+        scheduler_outputs=None,
+        model_output=model_output,
+        finished_before=None,
+        skip=None,
+    )
+
+
+if __name__ == '__main__':
+    parser = FlexibleArgumentParser(description="Benchmark per-batch.")
+    parser = AsyncEngineArgs.add_cli_args(parser)
+    vllm_args = ("--model meta-llama/Llama-3.1-8B-Instruct "
+                 f"-tp {1} "
+                 f"-pp {1} "
+                 "--collect-detailed-traces worker").split()
+    vllm_args = parser.parse_args(vllm_args)
+    vllm_args.max_model_len = 10000
+
+    benchmark_batch_param = BenchmarkBatchParam(
+        prefill_input_lens=[1024, 1024],
+        decode_input_lens=[128 for _ in range(512)],
+        log_dir='./logs',
+        gpu_freq_mhz=nvml_get_available_freq()[0],
+        delay_time_s=0.0,
+    )
+
+    uvloop.run(benchmark_batch(vllm_args, [benchmark_batch_param]))
```

Comment:
Could you not make a duplicate of `benchmark_batch.py`? So that when we need to change it later, we don't need to modify two places.